File: 백준 2003 수들의 합2/Main.java
import java.util.Scanner;

public class Main {
    private static int[] num;
    private static int left = 0;
    private static int right = 0;
    private static int result = 0;
    private static int sum = 0;

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int m = sc.nextInt();
        num = new int[n];
        for (int i = 0; i < num.length; i++) {
            num[i] = sc.nextInt();
        }
        while (true) {
            if (sum >= m) {
                sum -= num[left++];
            }else if(right == n){
                break;
            } else {
                sum += num[right++];
            }
            if (sum == m) {
                result++;
            }
        }
        System.out.println(result);


    }
}

File: 백준 2003 수들의 합2/Main.kt
import java.util.*

private lateinit var num: IntArray
private var left = 0
private var right = 0
private var result = 0
private var sum = 0


fun main(args: Array<String>) {
    val sc = Scanner(System.`in`)
    val n = sc.nextInt()
    val m = sc.nextInt()
    num = IntArray(n)
    for (i in num.indices) {
        num[i] = sc.nextInt()
    }
    while (true) {
        if (sum >= m) {
            sum -= num[left++]
        } else if (right == n) {
            break
        } else {
            sum += num[right++]
        }
        if (sum == m) {
            result++
        }
    }
    println(result)
}